cores/arduino/time.cpp

```diff
@@ -43,10 +53,16 @@ unsigned long millis()
 }
 
 unsigned long micros() {
-
-	// Convert time to us
+	// Return time in us
 	NVIC_DisableIRQ(main_timer.get_cfg()->cycle_end_irq);
-	uint32_t time_us = ((main_timer.get_period_raw() - main_timer.get_counter()) * 1000 / main_timer.get_period_raw()) +  (agt_time_ms * 1000);
+	uint32_t ms = agt_time_ms;
+	uint32_t const down_counts = _timer_get_counter();
```

Comment:
No macro, no inline function, directly accessing the registers.
```suggestion
	uint32_t const down_counts = R_AGT0->AGT; /* Obtain the current count value of the timer. */
```